jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java

```diff
@@ -701,74 +701,41 @@ protected void removeThread(Thread thread)
-                    break;
-                }
-
-                if ("select".equals(t.getMethodName()) && t.getClassName().endsWith("SelectorProducer"))
-                {
-                    knownMethod = "SELECTING ";
-                    break;
-                }
-
-                if ("accept".equals(t.getMethodName()) && t.getClassName().contains("ServerConnector"))
-                {
-                    knownMethod = "ACCEPTING ";
-                    break;
-                }
-            }
-            final String known = knownMethod;
-
-            if (isDetailedDump())
-            {
-                threads.add(new Dumpable()
-                {
-                    @Override
-                    public void dump(Appendable out, String indent) throws IOException
-                    {
-                        if (StringUtil.isBlank(known))
-                            Dumpable.dumpObjects(out, indent, String.format("%s %s %s %d", thread.getId(), thread.getName(), thread.getState(), thread.getPriority()), (Object[])trace);
-                        else
-                            Dumpable.dumpObjects(out, indent, String.format("%s %s %s %s %d", thread.getId(), thread.getName(), known, thread.getState(), thread.getPriority()));
-                    }
-
-                    @Override
-                    public String dump()
-                    {
-                        return null;
-                    }
-                });
-            }
+            StackTraceElement[] trace = thread.getStackTrace();
+            String known = getKnownMethod(trace);
+            String baseThreadInfo = String.format("%s %s %s %d", thread.getId(), thread.getName(), thread.getState(), thread.getPriority());
+
+            if (!StringUtil.isBlank(known))
+                threads.add(baseThreadInfo + " " + known);
+            else if (isDetailedDump())
+                threads.add((Dumpable)(o, i) -> Dumpable.dumpObjects(o, i, baseThreadInfo, (Object[])trace));
             else
-            {
-                int p = thread.getPriority();
-                threads.add(thread.getId() + " " + thread.getName() + " " + known + thread.getState() + " @ " + (trace.length > 0 ? trace[0] : "???") + (p == Thread.NORM_PRIORITY ? "" : (" prio=" + p)));
-            }
+                threads.add(baseThreadInfo + " @ " + (trace.length > 0 ? trace[0].toString() : "???"));
         }
 
+        DumpableCollection threadsDump = new DumpableCollection("threads", threads);
         if (isDetailedDump())
-        {
-            List<Runnable> jobs = new ArrayList<>(getQueue());
-            dumpObjects(out, indent, new DumpableCollection("threads", threads), new DumpableCollection("jobs", jobs));
-        }
+            dumpObjects(out, indent, threadsDump, new DumpableCollection("jobs", new ArrayList<>(getQueue())));
         else
+            dumpObjects(out, indent, threadsDump);
+    }
+
+    private String getKnownMethod(StackTraceElement[] trace)
```

Comment:
Should not this be `getKnownState()` as it returns the state of the thread?

Reply:
I kinda like the `getKnownMethod` name.
It's not returning the [`java.lang.Thread.State`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Thread.State.html).
It's intuiting the jetty specific state based on the method the stacktrace is sitting in.

Reply:
Ok so it's a string tag that indicates the Jetty thread stack state.
How about `getStateTag()` or `getCompressedStackTag()`?

Reply:
Or `getStackBehavior()` ?